### PR TITLE
Solution for problem 30 of Euler Project

### PR DESCRIPTION
**Pull request by deadshotsb**
Solution for problem 30 of Euler Project

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

**Comment by gaotian04** [comment]
。。

File: project_euler/problem_30/soln.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def digitsum(s):
+    c=0
+    for j in range(len(s)):
+        c+=pow(int(s[j]),5)
+    if c==int(s):
+        return c
+    else:
+        return 0
```
Comment (cclauss) [inline project_euler/problem_30/soln.py]:
```python
def digitsum(s: str) -> int:
    i = sum(pow(int(c), 5) for c in s)
    return i if i == int(s) else 0
```

File: project_euler/problem_30/soln.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+count=0
+for i in range(1000,1000000):
+    count+=digitsum(str(i))
```
Comment (cclauss) [inline project_euler/problem_30/soln.py]: # Replace 3 lines with one using the Python builtin __sum()__.
`count = sum(digitsum(str(i)) for i in range(1000,1000000))`

**Comment by cclauss** [comment]
My sense is that this is not the solution to Euler 30.  @onlinejudge95 @poyea Your assessments?

**Pull request merged**